### PR TITLE
Adding very naive (dumb) ingest using v2 api

### PR DESCRIPTION
**Pull request by mieciu**
Very rough implementation of the ingest endpoints with  `v2` api, but I think we should merge this and continue working on such "sketchpad". It basically shows the way.

### CODE REVIEW

File: quesma/frontend_connectors/elasticsearch_ingest.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+import (
+	"context"
+	"fmt"
+	"github.com/ucarion/urlpath"
+	"net/http"
+	quesma_api "quesma_v2/core"
+)
+
+type ElasticsearchIngestFrontendConnector struct {
```
Comment (pdelewski) [inline quesma/frontend_connectors/elasticsearch_ingest.go:14]: I would suggest to embed `frontend_connectors.BasicHTTPFrontendConnector` here and implement only `ServeHttp` or only part of it delegating common stuff to base implementation

```suggestion
type ElasticsearchIngestFrontendConnector struct {
  frontend_connectors.BasicHTTPFrontendConnector
```

**Comment by pdelewski** [comment]
Overall as we discussed LGTM :)

**Comment by pdelewski** [comment]
I think we can merge it. The only thing I would suggest before merge is to embed `frontend_connectors.BasicHTTPFrontendConnector` as I suggested above. Later we can refine `ElasticBackendConnector` interface

**Comment by mieciu** [comment]
> I think we can merge it. The only thing I would suggest before merge is to embed `frontend_connectors.BasicHTTPFrontendConnector` as I suggested above. Later we can refine `ElasticBackendConnector` interface

https://github.com/QuesmaOrg/quesma/pull/1055/commits/973e3089c8cf83b336c372ad5082755ef63c2eec 😉

File: quesma/frontend_connectors/elasticsearch_ingest.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		},
+	}
+	router := NewHTTPRouter()
+	router.AddRoute(IndexBulkPath, bulk)
+	router.AddRoute(IndexDocPath, doc)
+	fc.AddRouter(router)
+	return fc
+}
+
+func (h *ElasticsearchIngestFrontendConnector) AddRouter(router quesma_api.Router) {
```
Comment (pdelewski) [inline quesma/frontend_connectors/elasticsearch_ingest.go:43]:
```suggestion

```

File: quesma/frontend_connectors/elasticsearch_ingest.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	h.router = router
+}
+
+func (h *ElasticsearchIngestFrontendConnector) GetRouter() quesma_api.Router {
```
Comment (pdelewski) [inline quesma/frontend_connectors/elasticsearch_ingest.go:47]:
```suggestion

```

File: quesma/frontend_connectors/elasticsearch_ingest.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+		req.Header.Set("x-przemek", "blah")
+		_, message = dispatcher.Dispatch(handlerWrapper.Processors, metadata, message)
+		_, err := w.Write(message.([]byte))
+		if err != nil {
+			fmt.Printf("Error writing response: %s\n", err)
+		}
+	}).ServeHTTP(w, req)
+}
+
+func (h *ElasticsearchIngestFrontendConnector) Listen() error {
```
Comment (pdelewski) [inline quesma/frontend_connectors/elasticsearch_ingest.go:84]:
```suggestion

```

File: quesma/frontend_connectors/elasticsearch_ingest.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	h.listener.Handler = h
+	go func() {
+		err := h.listener.ListenAndServe()
+		_ = err
+	}()
+
+	return nil
+}
+
+func (h *ElasticsearchIngestFrontendConnector) Stop(ctx context.Context) error {
```
Comment (pdelewski) [inline quesma/frontend_connectors/elasticsearch_ingest.go:96]:
```suggestion

```

File: quesma/frontend_connectors/elasticsearch_ingest.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+		return nil
+	}
+	err := h.listener.Shutdown(ctx)
+	if err != nil {
+		return err
+	}
+	return h.listener.Close()
+}
+
+func (h *ElasticsearchIngestFrontendConnector) GetEndpoint() string {
```
Comment (pdelewski) [inline quesma/frontend_connectors/elasticsearch_ingest.go:107]:
```suggestion

```

File: quesma/frontend_connectors/elasticsearch_ingest.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+	IngestAction    = "ingest_action"
+	DocIndexAction  = "_doc"
+	BulkIndexAction = "_bulk"
+	IngestTargetKey = "ingest_target"
+	// TODO: this actually should not be a dependency on processor
+)
+
+func NewElasticsearchIngestFrontendConnector(endpoint string) *ElasticsearchIngestFrontendConnector {
+	fc := &ElasticsearchIngestFrontendConnector{
+		BasicHTTPFrontendConnector: BasicHTTPFrontendConnector{
```
Comment (pdelewski) [inline quesma/frontend_connectors/elasticsearch_ingest.go:32]: As we are embedding `BasicHTTPFrontendConnector` below methods should not be needed and can be removed

Comment (mieciu) [reply quesma/frontend_connectors/elasticsearch_ingest.go:31]: https://github.com/QuesmaOrg/quesma/pull/1055/commits/e2ec36bbdc86eee719142405de97489a5061fa4d

**Review by pdelewski — APPROVED** [review]

**Pull request merged**